### PR TITLE
Add discrete-token ASR for LibriSpeech 100h

### PR DESCRIPTION
**Pull request by akreal**
## What?

Discrete-token ASR for LibriSpeech 100h.

## Why?

There is no discrete-token ASR recipe for this small scale dataset yet.

### CODE REVIEW

**Comment by sw005320** [comment]
Very cool!
This result is very good compared with our **initial** trial in https://arxiv.org/pdf/2305.18108.pdf.

File: egs2/librispeech_100/asr2/conf/train_discrete_asr_e_branchformer1_1gpu.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Trained with A100 (40 GB) x 1 GPUs for Kmeans1K+nbpe5K. It takes 32 minutes per epoch.
```
Comment (simpleoier) [inline egs2/librispeech_100/asr2/conf/train_discrete_asr_e_branchformer1_1gpu.yaml:1]: Can you update this information?

File: egs2/librispeech_100/asr2/run.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+src_nbpe=6000   # I use src_nbpe=6000 for 2000-cluster kmeans.
+tgt_nbpe=5000   # if token_joint is True, then only tgt_nbpe is used
+
+# ts: true sequence
+# rm: deduplicated sequence which removes duplicated tokens
+src_case="rm"
+tgt_case="ts"
+
+./asr2.sh \
+    --kmeans_opts "--batch_bins 4800000" \
```
Comment (simpleoier) [inline egs2/librispeech_100/asr2/run.sh:31]: Can you reduce the number of gpu jobs in `kmeans_opts`: adding an argument `--nj 4` for example?

**Review by simpleoier — COMMENTED** [review]
Thank you! The results are very cool.
I left two minor comments.

File: egs2/librispeech_100/asr2/conf/train_discrete_asr_e_branchformer1_1gpu.yaml (hunk 1 above)
Comment (akreal) [reply egs2/librispeech_100/asr2/conf/train_discrete_asr_e_branchformer1_1gpu.yaml]: Done!

File: egs2/librispeech_100/asr2/run.sh (hunk 2 above)
Comment (akreal) [reply egs2/librispeech_100/asr2/run.sh]: Done!

**Review by simpleoier — APPROVED** [review]
Thanks!

**Pull request merged**